osf/metrics.py

```diff
@@ -196,6 +197,60 @@ class Index:
             'refresh_interval': '1s',
         }
 
+    class Meta:
+        source = metrics.MetaField(enabled=True)
+
+    @classmethod
+    def filter_institution(cls, institution):
+        return cls.search().filter('match', institution_id=institution._id)
+
+    @classmethod
+    def get_department_counts(cls, institution) -> list:
+        """
+        Gets the most recent document for every unique user.
+        :param institution: Institution
+        :return: list
+        """
+        search = cls.filter_institution(institution).sort('timestamp')
+        yesterday = dt.date.today() - dt.timedelta(days=1)
+
+        search.update_from_dict({
+            'aggs': {
+                'date_range': {
+                    'filter': {
+                        'range': {
+                            'timestamp': {
+                                'gte': str(yesterday),
+                            }
+                        }
+                    },
+                    'aggs': {
+                        'departments': {
+                            'terms': {
+                                'field': 'department',
+                                'missing': 'N/A',
+                                'size': 250
+                            },
+                            'aggs': {
+                                'users': {
+                                    'terms': {
+                                        'field': 'user_id'
```

Comment:
i get a 500 error for this query locally:
> elasticsearch.exceptions.RequestError: RequestError(400, 'search_phase_execution_exception', 'Fielddata is disabled on text fields by default. Set fielddata=true on [user_id] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead.')

checked out the mapping for the `user_id` field:
```
          "user_id" : {
            "type" : "text",
            "fields" : {
              "keyword" : {
                "type" : "keyword",
                "ignore_above" : 256
              }
            }
          }
```
and looks like, while `user_id` is a text field, there's also a `user_id.keyword` field that we could use instead.

this change fixed the error for me
```suggestion
                                        'field': 'user_id.keyword'
```
but i'm also curious why this happened for me but not you -- what's your index mapping like? (see http://localhost:9201/osf_userinstitutionprojectcounts_2020/?pretty)

Reply:
disregard, this was my local mappings being out-of-date -- in the current mappings `user_id` is of type `keyword`